stage1/enter/enter.c

```diff
@@ -50,17 +51,48 @@ static int openpidfd(int pid, char *which) {
 int main(int argc, char *argv[])
 {
 	int	fd;
-	int	pid;
+	int	pid = 0;
+	char *appname = NULL;
 	pid_t	child;
 	int	status;
 	int	root_fd;
 
+	int c;
+
 	/* The parameters list is specified in
 	 * Documentation/devel/stage1-implementors-guide.md */
-	exit_if(argc < 4,
-		"Usage: %s pid appName cmd [args...]", argv[0])
+	while (1) {
+		int option_index = 0;
+		static struct option long_options[] = {
+		   {"pid",     required_argument, 0,  'p' },
+		   {"appname", required_argument, 0,  'a' },
+		   {0,         0,                 0,  0 }
+		};
+
+		c = getopt_long(argc, argv, "p:a:",
+				long_options, &option_index);
+		if (c == -1)
+			break;
+
+		switch (c) {
+		case 'p':
+			pid = atoi(optarg);
+			break;
+		case 'a':
+			appname = optarg;
+			break;
+		case 0:
+			break;
+		case ':':   /* missing option argument */
+		case '?':
+		default:
+			fprintf(stderr, "Usage: %s --pid=1234 "
+					"--appname=name -- cmd [args...]",
+					argv[0]);
+			exit(1);
+		}
```

Comment:
I thought it would help with the "stable and documented stage0/stage1 ABI" item on the [roadmap for 0.8.0](https://github.com/coreos/rkt/blob/master/ROADMAP.md#rkt-08-august).

The motivation is on issue https://github.com/coreos/rkt/issues/1133 and the parameters for enter are described on: [Stage1 ACI implementor's guide](https://github.com/coreos/rkt/blob/master/Documentation/devel/stage1-implementors-guide.md)